### ISSUE TITLE
Introduce a new GitHub Action to use, or use the one we already use?

### ISSUE THREAD

**Issue by consideRatio** [sgibson91/helm-upgrade-decision-logic#1]
I note that another github action has been used to get changed paths. Did you know that the one we use in 2i2c/infrastructure [can emit changed files](https://github.com/dorny/paths-filter#outputs) as well?

For reference, this repo now introduces https://github.com/jitterbit/get-changed-files, while 2i2c/infrastructure use https://github.com/dorny/paths-filter#outputs.

**Comment by sgibson91** [comment]
I did know that. But I like the one I'm using here as we automatically get outputs related to _all_ changes, included deleted files. So this could pave the way to automatically deleting hubs when config is removed if we want.

**Comment by sgibson91** [comment]
We also have to define the filepaths twice using the dorny paths action, which I've come to not like.

**Comment by consideRatio** [comment]
> But I like the one I'm using here as we automatically get outputs related to all changes.

Oh, and the other didn't list the deleted files or didn't even trigger on deleted files?! That was an unexpected behavior to me and worth reporting as a bug. I didn't find a bug report about this.

> We also have to define the filepaths twice using the dorny paths action, which I've come to not like.

If you want to use it as a way to get the changed files, you could do...

```
      - uses: dorny/paths-filter@v2
        id: changed-files
        with:
          filters: |
            any: ["**"]
```

I'd guess this could be quite easy with a `git` command as well, give two references to compare with.

Anyhow to resolve this issue, I think everything is fine but I'd appreciate having a note that motivates using another file path action. I can submit a PR to add a comment about the motivation you give in this issue.

**Comment by sgibson91** [comment]
Some context for this discussion:

- I wrote the original workflow file we're talking about
- I **only** knew about dorny/paths-filter at the time I wrote it, and learned about jitterbit/get-changed-files afterwards
- Working on personal projects, I find jitterbit's action easier to set up. I only need to configure the action triggers itself and the action does hard work for me. And I prefer the range of default outputs we get from jitterbit's action (quoted below) which I believe are more fine-grained than dorny's and would open pathways to other kinds of deploy logic should we wish.

>  The steps output context exposes the output names `all`, `added`, `modified`, `removed`, `renamed`, and `added_modified`.
